### PR TITLE
std::vector fixes

### PR DESCRIPTION
**Pull request by TurkeyMan**
Found some edge cases I missed in the tests.

### CODE REVIEW

**Comment by TurkeyMan** [comment]
Wait up, I just found one more while this was running the tests... let me sleep on this a couple of nights.
There's a few curly little edge cases with construction and stuff.

**Comment by TurkeyMan** [comment]
`@disable` default constructor caused issues in this PR, but there's a copy construction issue too; and https://issues.dlang.org/show_bug.cgi?id=19902 is a problem >_<

**Comment by thewilsonator** [comment]
OK, ping when when you're happy with it.

**Comment by TurkeyMan** [comment]
Okay, I'm done with this for the moment.
I powered up the tests, worked through the `@disable` ctor stuff, and removed the `alias this` (replaced with all the slice operators).

File: test/stdcpp/src/vector.cpp [hunk 1]
```diff
@@ -1,17 +1,60 @@
...
+    }
+    NotPOD(const NotPOD& rh)
+    {
+        for (int i = 0; i < 5; ++i)
+            x[i] = rh.x[i];
+        wasInit = rh.wasInit;
+        if (rh.wasInit)
+            ++vectorTest_numAllocated;
+    }
+    NotPOD(NotPOD&& rh) noexcept
```
Comment (thewilsonator) [inline test/stdcpp/src/vector.cpp]: `test\stdcpp\src\vector.cpp(23) : error C3646: 'noexcept' : unknown override specifier`

File: test/stdcpp/src/vector.cpp [hunk 2]
```diff
@@ -1,17 +1,60 @@
...
+        wasInit = rh.wasInit;
+        rh.wasInit = false;
+    }
+    ~NotPOD()
+    {
+        if (wasInit)
+            --vectorTest_numAllocated;
+    }
+
+    int x[5] = { 1, 2, 3, 4, 5 };
```
Comment (thewilsonator) [inline test/stdcpp/src/vector.cpp]: `test\stdcpp\src\vector.cpp(36) : error C2536: 'NotPOD::NotPOD::x' : cannot specify explicit initializer for arrays`

Comment (TurkeyMan) [reply test/stdcpp/src/vector.cpp]: yep, bloody ancient compilers!
